navbar: Add user avatars with availability dots for DM conversations.

pranav718 opened the pull request:
Replace text-based names with clickable avatar images in the DM navbar.
Each avatar displays the user's presence status via a colored availability
dot. Clicking an avatar opens the user card popover.

Fixes: #37078

**How changes were tested:**
- Opened a 1-on-1 DM conversation, single avatar displayed
- Opened a group DM conversation, multiple avatars displayed  
- Clicked an avatar, user card popover opens correctly
- Verified availability dots show correct status (green for active, empty for offline)


**Screenshots and screen records**

Before: 

(1-on-1 DM)
<img width="839" height="312" alt="Screenshot 2025-12-16 at 4 22 57 PM" src="https://github.com/user-attachments/assets/2da93269-766a-4b19-8d99-eb800dafc034" />

(group DM)
<img width="839" height="298" alt="Screenshot 2025-12-16 at 4 26 27 PM" src="https://github.com/user-attachments/assets/0b2f6d84-1ca4-414b-9aba-9c75401239ef" />




After:

(both 1-on-1 DM and group DM)

https://github.com/user-attachments/assets/f073dc76-4980-457e-a207-4d4479420f93

**Self-review checklist:**
- [x] I have tested this change on a Zulip development environment.
- [x] I have included screenshots/recordings of the changes (for UI changes).
- [ ] I have written/updated automated tests for the new behavior.
- [ ] I have documented any new/modified endpoints in [api_docs/changelog.md](cci:7://file:///Users/pranavray/zulip/api_docs/changelog.md:0:0-0:0).
- [ ] I have documented any new/modified CSS classes in `web/styles/README.md`.

pranav718 commented:
### Implementation Notes

To implement the clickable avatar to user card popover functionality, I explored several approaches to handle the import between `message_view_header.ts` and `user_card_popover.ts`:

1. **Direct import** - Created circular import dependencies
2. **Dynamic import with async** - ESLint's `import/no-cycle` still tracked it
3. **Dynamic import inside setTimeout** - ESLint still detected the cycle

**Final solution:** Custom DOM event pattern
- `message_view_header.ts` dispatches `dm-navbar-avatar-click` custom event
- `user_card_popover.ts` listens for that event in its `initialize()` function

This creates zero direct import relationship between the files while maintaining full functionality.

Happy to address any feedback or suggestions!

alya commented:
Thanks for giving it a try! I'll go ahead and close this PR, as it doesn't follow the expected visuals, and randomly introduces circular avatars instead.

pranav718 commented:
Hi @alya, thanks for the feedback! I've addressed the circular avatar issue,  changed to `border-radius: 4px` to match Zulip's standard avatar styling (square with rounded corners). 

Could I reopen this PR or submit a new one with the corrected visuals? Happy to make any other adjustments needed to match the expected design.

alya commented:
You can reopen this one if you like.
